dbt/include/duckdb/macros/materializations/external.sql

```diff
@@ -7,11 +7,14 @@
   {%- set read_location = adapter.external_read_location(location, rendered_options) -%}
 
   -- set language - python or sql
+  -- i have to learn general about python models
   {%- set language = model['language'] -%}
 
   {%- set target_relation = this.incorporate(type='view') %}
 
   -- Continue as normal materialization
+  -- Why do we have temp and intermediate relation? 
```

Comment:
The idea is to not destroy an existing relation that has the same name as this one (if it exists in the catalog) in case something fails during the construction of the relation defined by this model.

Reply:
i understand; we prepare everything and do all the transformation, and at the very end, we exchange the names

I understand how it fits into the context but am not sure what it means for me right now in the implementation, so make sure that i don't overlook that one in the implementation.